### PR TITLE
Reserve hash routes for codes

### PR DESCRIPTION
**Pull request by JustusFT**
OP#1959

### CODE REVIEW

**Review by donpui — APPROVED** [review]
Looks, will need to test it properly when deployed

File: client/src/app/hooks/useCodeUrlCheck.ts [hunk 1]
```diff
@@ -11,15 +11,24 @@ export function useCodeUrlCheck() {
...
-      codeInput?.setValue(code);
-      codeInput?.setSubmitting(true);
+    if (location.pathname == "/" && location.hash) {
       navigate("/r", { replace: true });
-      // HACK: have a better way to wait for wormhole to initialize
-      setTimeout(() => {
-        wormhole?.saveFile(code);
-      }, 2000);
+
+      const code = location.hash.slice(1);
```
Comment (donpui) [inline client/src/app/hooks/useCodeUrlCheck.ts:17]: I think we need slide 2 symbols as I copy paste url http://localhost:8080/#/3-atlantic-crucial, I get
image

File: client/src/app/wormhole/client_worker.ts [hunk 2]
```diff
@@ -164,7 +164,7 @@ export default class ClientWorker implements ClientInterface {
     id,
     error,
   }: RPCMessage): Promise<void> {
-    window.history.pushState({}, "", "/#/s?cancel=");
+    window.history.pushState({}, "", "/s?cancel=");
```
Comment (donpui) [inline client/src/app/wormhole/client_worker.ts:167]: Also If I upload file and cancel, it redirect to receiver:
image

File: client/src/app/hooks/useCodeUrlCheck.ts (hunk 1 above)
Comment (JustusFT) [reply client/src/app/hooks/useCodeUrlCheck.ts:17]: The last `/` has been dropped. The url you receive should be http://localhost:8080/#3-atlantic-crucial now.

**Comment by donpui** [comment]
Now only fails with cancellation on sender and receiver sides:
Cannot GET /s
Cannot GET /r

**Pull request merged**